src/asmcnc/apps/systemTools_app/screen_manager_systemtools.py

```diff
@@ -145,6 +147,124 @@ def get_grbl_settings_from_usb():
+                    os.system('cp /home/pi/multiply.txt /home/pi/easycut-smartbench/transfer_tmp/')
+                    os.system('cp /home/pi/plus.txt /home/pi/easycut-smartbench/transfer_tmp/')
+                    # compress everything to usb
+                    print('Create tar...')
+                    os.system('sudo tar czf /media/usb/transfer.tar.gz -C /home/pi/easycut-smartbench/transfer_tmp .')
+                except:
+                    pass
+                try:
+                    #clean up
+                    print('Clean up...')
+                    os.system('rm -r /home/pi/easycut-smartbench/transfer_tmp')
+                except:
+                    pass
+                wait_popup.popup.dismiss()
+                #check if transfer file exists
+                if os.path.isfile('/media/usb/transfer.tar.gz'):
+                    message = self.l.get_str('Successfully saved settings to USB!')
+                    popup_info.PopupMiniInfo(self.sm, self.l, description=message)
+                else:
+                    message = self.l.get_str('Could not save settings. Please check USB!')
+                    popup_info.PopupMiniInfo(self.sm, self.l, description=message)
+                self.usb_stick.disable()
+                self.mutex.release()
+            elif loop_counter > 30:
+                wait_popup.popup.dismiss()
+                message = self.l.get_str('No USB found!')
+                popup_info.PopupMiniInfo(self.sm, self.l, description=message)
+                self.mutex.release()
+            else:
+                Clock.schedule_once(lambda dt: copy_settings_to_usb(loop_counter+1), 0.2)
+
+        Clock.schedule_once(lambda dt: copy_settings_to_usb(1), 0.2)
+
+    '''Restores all the relevant files from USB for upgrading the console to a c10 .'''
+    def show_popup_before_upload_settings_from_usb(self):
+        self.sm.pm.show_upload_settings_popup(self)
+
+    def upload_settings_from_usb(self, *args):
+        if self.mutex.locked():
+            # already running
+            print('mutex locked!')
+            return
+        self.mutex.acquire(True)
+        self.usb_stick.enable()
+        message = self.l.get_str('Restoring settings from USB. Please wait') + '...'
+        wait_popup = popup_info.PopupWait(self.sm, self.l, description=message)
+
+        def restore_settings_from_usb(loop_counter):
+            print('Loop: {}').format(loop_counter)
+            if self.usb_stick.is_usb_mounted_flag:
+                # TODO restore files here
+                success_flag = True
+                try:
+                    if not os.path.isfile('/media/usb/transfer.tar.gz'):
+                        wait_popup.popup.dismiss()
+                        self.usb_stick.disable()
+                        message = self.l.get_str('Could not restore Settings. Please check USB!')
+                        popup_info.PopupMiniInfo(self.sm, self.l, description=message)
```

Comment:
this old call to this popup should be changed to new system, see popup_manager for function

Reply:
resolved